app/src/main/java/app/revanced/integrations/youtube/patches/spoof/SpoofClientPatch.java

```diff
@@ -79,201 +86,89 @@ public static String blockInitPlaybackRequest(String originalUrlString) {
     /**
      * Injection point.
      */
-    public static int getClientTypeId(int originalClientTypeId) {
-        return SPOOF_CLIENT_ENABLED ? SPOOF_CLIENT_TYPE.id : originalClientTypeId;
-    }
-
-    /**
-     * Injection point.
-     */
-    public static String getClientVersion(String originalClientVersion) {
-        return SPOOF_CLIENT_ENABLED ? SPOOF_CLIENT_TYPE.appVersion : originalClientVersion;
+    public static boolean isSpoofingEnabled() {
+        return SPOOF_STREAMING_DATA;
     }
 
     /**
      * Injection point.
      */
-    public static String getClientModel(String originalClientModel) {
-        return SPOOF_CLIENT_ENABLED ? SPOOF_CLIENT_TYPE.model : originalClientModel;
-    }
+    public static UrlRequest buildRequest(UrlRequest.Builder builder, String url,
+                                          Map<String, String> playerHeaders) {
+        if (SPOOF_STREAMING_DATA) {
+            try {
+                Uri uri = Uri.parse(url);
+                String path = uri.getPath();
+                if (path != null && path.contains("player") && !path.contains("heartbeat")) {
```

Comment:
Why is heartbeat checked here?

Reply:
`/player/heartbeat` doesn't have `id` parameter, so i just excluded it.